microservices/front-end/src/components/MainQuestionBank.tsx

```diff
@@ -131,8 +158,72 @@ const QuestionBank: React.FC = () => {
     }
 
     setQuestions(fetchedQuestions);
+    return fetchedQuestions; // return the questions
   };
 
+  const setQuestionSummary = async (questions?: Question[]) => {
+    if (questions && questions.length > 0) {
+      const summary = calculateCategorySummary(questions);
+      setCategorySummary(summary);
+    } else {
+      alert("Failed to fetch question summary");
+    }
+  };
+
+  // dummy function for filter changes
+  const handleAttemptFilterChange = (attempted: string) => {
+    setAttemptedFilter(attempted);
+  };
+
+  const handleSearch = (query: string) => {
+    setSearchQuery(query);
+  };
+
+  const filterQuestions = (
+    questions: Question[],
+    difficulty: string,
+    category: string,
+    query: string,
+    attempted: string
+  ): Question[] => {
+    return questions.filter(
+      (question) =>
+        (difficulty === "All" || question.complexity === difficulty) &&
+        (category === "All" || question.categories.includes(category)) &&
+        (query === "" ||
+          question.title.toLowerCase().includes(query.toLowerCase())) &&
+        (attempted === "All" ||
+          (attempted === "Attempted" &&
+            attemptedQuestions.includes(question._id)) ||
+          (attempted === "Unattempted" &&
+            !attemptedQuestions.includes(question._id)))
```

Comment:
Please refactor this part to use multiple variables with descriptive names instead of just chaining boolean conditions together